core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -358,8 +361,11 @@ class DefaultMessageFormatter extends MessageFormatter {
   }
 
   def writeTo(key: Array[Byte], value: Array[Byte], timestamp: Long, timestampType: TimestampType, output: PrintStream) {
-    if (timestampType != TimestampType.NO_TIMESTAMP_TYPE) {
-      output.write(s"$timestampType:$timestamp".getBytes)
+    if (printTimestamp) {
+      if (timestampType != TimestampType.NO_TIMESTAMP_TYPE)
+        output.write(s"$timestampType:$timestamp".getBytes)
+      else
+        output.write(s"NO_TIMESTAMP".getBytes)
```

Comment:
Should we write out the separator after the timestamp? Maybe we can just piggyback on keySeparator.